src/main/java/frc/robot/subsystems/swerve/PhoenixOdometryThread.java

```diff
@@ -38,14 +42,36 @@
  * "waitForAll" blocking method to enable more consistent sampling. This also allows Phoenix Pro
  * users to benefit from lower latency between devices using CANivore time synchronization.
  */
-public class PhoenixOdometryThread extends Thread {
-  public record Registration(ParentDevice device, Set<StatusSignal<Double>> signals) {}
+public class PhoenixOdometryThread extends Thread implements OdometryThreadIO {
+  public enum SignalType {
+    DRIVE,
+    STEER,
+    GYRO;
+    // In theory we could measure other types of info in this thread
+    // But we don't!
+  }
+
+  public record SignalID(SignalType type, int modID) {}
+
+  public record Registration(
+      ParentDevice device,
+      Optional<ModuleConstants> moduleConstants,
+      SignalType type,
+      Set<StatusSignal<Double>> signals) {}
+
+  public record RegisteredSignal(
+      StatusSignal<Double> signal, Optional<ModuleConstants> moduleConstants, SignalType type) {}
 
-  public record Samples(double timestamp, Map<StatusSignal<Double>, Double> values) {}
+  public record Samples(double timestamp, Map<SignalID, Double> values) {}
 
   private final ReadWriteLock journalLock = new ReentrantReadWriteLock(true);
 
-  private final Set<StatusSignal<Double>> signals = Sets.newHashSet();
+  // For gyros
+  private static final ModuleConstants NEGATIVE_ONE =
+      new ModuleConstants(-1, "", -1, -1, -1, Rotation2d.fromRotations(0));
```

Comment:
we do use them as integers for indexing the module array as part of `updateOdometry`. It would be trivial to cast a `String` there, but i feel like casting is inviting errors vs just having a constant for gyros to special case that signal.

Reply:
missing rust enums for some of this, would be nice to have `DRIVE` and `STEER` contain an id while `GYRO` doesnt